src/hash.rs

```diff
@@ -31,3 +31,30 @@ impl<'de> Visitor<'de> for HashString {
         }
     }
 }
+
+/// A visitor that deserializes a public offer
+#[cfg(feature = "serde")]
+pub(crate) struct OfferString;
+
+#[cfg(feature = "serde")]
+impl<'de> Visitor<'de> for OfferString {
+    type Value = String;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            formatter,
+            "a string representing a public offer in Base58 value"
+        )
+    }
+
+    fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        if s.len() == 195 {
```

Comment:
Are we sure that the len is always 195, my take is not as base58 can vary depending on values

Reply:
No, not sure about this - are there any variable size members of `PublicOffer`?

Reply:
The way it currently works yes, depending on the type of assets you may have different bytes serialized (but with only xmr/btc we are fine so we can keep it).